data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -107,14 +106,15 @@ private void translateSource(Object sourceObject, Event recordEvent, TargetsPara
         }
 
         String rootField = jsonExtractor.getRootField(commonPath);
-        if(!recordObject.containsKey(rootField)){
+        Object rootFieldObject = recordEvent.get(rootField, Object.class);
```

Comment:
Is this faster? If so, can we optimize `containsKey` instead of making this change?

Reply:
Biggest performance issue is calling `toMap()`. I wanted to avoid calling until it is actually needed. Worked with a customer provided 50000 line yaml and this change seems to be helping.

Reply:
So is change to `get` doing anything to help with performance? If not, I'd suggest reverting to `containsKey`. If so, let's fix `containsKey` in JacksonEvent and then revert to use `containsKey` here.

Reply:
@dlvenable the profiler showed that a lot of time was spent in `event.toMap()`  Not sure how fixing containsKey would help here.

Reply:
Data Prepper `Event` has a `containsKey()` method.

```
if(!recordEvent.containsKey(rootField)) {
...
```

Reply:
Oh Ok. So, you are referring to line 110 and not 92. I did not try with `containsKey`. I can try that and get back to you.

Reply:
@dlvenable looking at the code, both get() and containsKey() code is almost same. Definitely using containsKey would help even more.